### ISSUE TITLE
MeshInstance3d Warning not Disappearing after correcting Solutions suggested

### ISSUE THREAD

**Issue by pixelinkmedia** [godotengine/godot#85710]
### Godot version

v4.2.stable.official [46dc27791]

### System information

Windows 10 Home

### Issue description

I converted a project from 3.2 to 4.0.1 (where issue started) and then tried 4.2.
I got a warning about MeshInstant3d's Mesh being deprecated. I resaved the scene, deleted both MeshInstant3D nodes, and recreated the default "New Box Mesh".... cleared the warning. Reran the project and the warning still says the same thing. Not sure if its a bug or what?

### Steps to reproduce

see above comments... it contains the steps

### Minimal reproduction project

I have a screenshot in the root folder called... "screenshotofMeshInstant3d_Issue.png"

[BallMaze.zip](https://github.com/godotengine/godot/files/13539566/BallMaze.zip)

**Comment by pixelinkmedia** [comment]
No, I used the default "BoxMesh" as I stated above. It works, it just the warning doesn't disappear.

**Comment by AThousandShips** [comment]
Did you replace all? You have several meshes here and a scene that might contain one, are you sure that's the mesh instance that has the warning?

Try reimporting all your meshes and then see if the warning shows

**Comment by pixelinkmedia** [comment]
Yes, did you look at the screenshot?
I said I deleted BOTH meshinstance3d nodes and recreated the box meshed.

**Comment by AThousandShips** [comment]
I did, are you saying the "Character" scene contains no meshes?

**Comment by smix8** [comment]
The error is from the ArrayMesh resource when an old "surfaces" array data is found, it is not part from the MeshInstance3D.

**Comment by AThousandShips** [comment]
This error doesn't highlight anything here, that's one of the limitations with this case

**Comment by pixelinkmedia** [comment]
Well, it did for me once.
Also, I reimported the 3d mesh, but still get the warning. There are no other meshed in the project

**Comment by pixelinkmedia** [comment]
Everything was in the zip. Even the 3d ball exported from blender. So, you can try it yourself.

**Comment by AThousandShips** [comment]
I am unable to test this right now as I'm not on my computer, but this warning does not point to the mesh instance, you must have misunderstood, as it isn't an warning that works that way

**Comment by clayjohn** [comment]
The error is coming from "res://Assets/playerball.tres" saving it is enough to remove the warning. 

BoxMeshes always use up-to-date formats and never need to be resaved.

**Comment by pixelinkmedia** [comment]
Okay, I resaved the TRES and the tSCN file.... still get warning.
I don't understand what it means to "save the path".... is there a setting for that because all I am doing is resaving the scene/trest
see image from warning...

![image](https://github.com/godotengine/godot/assets/51757955/bb56c543-f102-4f97-9d51-b661e0c355d1)

**Comment by AThousandShips** [comment]
Do you have any local meshes in any scene? The empty path sounds like that

**Comment by pixelinkmedia** [comment]
Character is the only imported mesh.... see tree.
![image](https://github.com/godotengine/godot/assets/51757955/993ba8f1-0679-40a0-8797-7d5c98f7b7dd)


I also have a light. forgot that.


This is all that is in the scene...
![image](https://github.com/godotengine/godot/assets/51757955/5b7aac0f-8bae-4909-b77a-8a33c4f3c208)

**Comment by clayjohn** [comment]
Here is a screenshot showing you how to save the mesh resource:
![image](https://github.com/godotengine/godot/assets/16521339/9ca30057-53d0-4c59-867b-a97fcb5d2779)

That is all I did in your MRP and the warning went away
